optaplanner-quarkus-integration/optaplanner-quarkus/runtime/src/main/java/org/optaplanner/quarkus/nativeimage/Substitute_MemberAccessorFactory.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.quarkus.nativeimage;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.lang.reflect.Member;
+import java.lang.reflect.Method;
+
+import org.optaplanner.core.impl.domain.common.ReflectionHelper;
+import org.optaplanner.core.impl.domain.common.accessor.MemberAccessor;
+import org.optaplanner.core.impl.domain.common.accessor.MemberAccessorFactory;
+import org.optaplanner.core.impl.domain.common.accessor.ReflectionBeanPropertyMemberAccessor;
+import org.optaplanner.core.impl.domain.common.accessor.ReflectionFieldMemberAccessor;
+import org.optaplanner.core.impl.domain.common.accessor.ReflectionMethodMemberAccessor;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@TargetClass(className = "org.optaplanner.core.impl.domain.common.accessor.MemberAccessorFactory")
+public final class Substitute_MemberAccessorFactory {
```

Comment:
Personally, I think that this code belongs in `optaplanner-core`. The reason it's not there, is that the annotations used prevent that. Simple solution: put the impl in core, but leave the annotating bells and whistles here.

Reply:
Also, I don't know why this class exists. Why substitute one part of code that does reflection, with another one that still does reflection?

Reply:
This is for Graal: it can't handle MethodHandles, but it can handle old-fashioned reflection.
Once our gizmo work starts, we can probably ditch MethodHandles alltogether (they are faster than old-fasioned reflection, but slower than gizmo code generation)

Reply:
This is definitely an open issue. I 've created a jira to track:
https://issues.redhat.com/browse/PLANNER-2002